builtin/logical/transit/path_sign_verify.go

```diff
@@ -373,13 +372,6 @@ func (b *backend) pathSignWrite(ctx context.Context, req *logical.Request, d *fr
 		return logical.ErrorResponse(fmt.Sprintf("key type %v does not support signing", p.Type)), logical.ErrInvalidRequest
 	}
 
-	// Allow managed keys to specify no hash algo without additional conditions.
-	if hashAlgorithm == keysutil.HashTypeNone && p.Type != keysutil.KeyType_MANAGED_KEY {
-		if !prehashed || sigAlgorithm != "pkcs1v15" {
```

Comment:
This inner if branch should be retained I believe.

Reply:
Thanks for the early feedback. Going to have a look. :eyes:

Reply:
Having the inner loop seems to be breaking the transit tests.

Reply:
To be clear, I think the full conditional should be:

```go
if hashAlgorithm == keysutil.HashTypeNone && (!prehashed || sigAlgorithm != "pkcs1v15") { ... }
```

right? The top conditional applied if the key type was _not_ a managed key, and thus should be retained with the assertion around None hash type.

Reply:
Of course 🤦.
With this change, the tests pass again.